elements/ethernet/arpfaker.cc

```diff
@@ -57,9 +63,18 @@ ARPFaker::initialize(ErrorHandler *)
 void
 ARPFaker::run_timer(Timer *)
 {
-    if (Packet *p = ARPResponder::make_response(_eth1.data(), _ip1.data(),
-						_eth2.data(), _ip2.data()))
-	output(0).push(p);
+    Packet *p = ARPResponder::make_response(
+        _eth1.data(), _ip1.data(), _eth2.data(), _ip2.data()
+    );
+
+    if (p) {
+    #if HAVE_BATCH
```

Comment:
This is not sufficient, as a path may be not batch-enabled, there is the in_batch_mode() function to check that